src/udev/net/link-config.c

```diff
@@ -236,16 +237,27 @@ bool link_config_should_reload(link_config_ctx *ctx) {
 }
 
 int link_config_get(link_config_ctx *ctx, sd_device *device, link_config **ret) {
+        struct ether_addr permanent_mac;
         link_config *link;
+        const char *name;
+        int r;
 
         assert(ctx);
         assert(device);
         assert(ret);
 
+        r = sd_device_get_sysname(device, &name);
+        if (r < 0)
+                return r;
+
+        r = ethtool_get_permanent_macaddr(&ctx->ethtool_fd, name, &permanent_mac);
+        if (r < 0)
+                log_device_debug_errno(device, r, "Failed to get permanent MAC address, ignoring: %m");
```

Comment:
if this fails (which I think is likely on devices without ROM?) then permanent_mac is not initialized here, and we end up comparing an unitinitialized field with the matches we are looking for.

Thus I think we should explicitly initialized permanent_mac to `{}` before we call ethtool_get_permanent_macaddr() so we know its nitialized and then I figure net_match_config() should probably explicitly decide that no permanent mac match ever matches an all-zero MAC address.